core/backend/src/ViewStateHydrater.ts

```diff
@@ -0,0 +1,126 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { CompressedId64Set, Id64String } from "@itwin/core-bentley";
+import { HydrateViewStateRequestProps, HydrateViewStateResponseProps, ModelProps, QueryRowFormat, SubCategoryResultRow, ViewAttachmentProps, ViewStateLoadProps } from "@itwin/core-common";
+import { IModelDb } from "./IModelDb";
+
+export class ViewStateHydrater {
+  private _imodel: IModelDb;
+  public constructor(iModel: IModelDb) {
+    this._imodel = iModel;
+  }
+
+  public async getHydrateResponseProps(options: HydrateViewStateRequestProps): Promise<HydrateViewStateResponseProps> {
+    const response: HydrateViewStateResponseProps = {};
+    const promises = [];
+    if (options.acsId)
+      promises.push(this.handleAcsId(response, options.acsId));
+    if (options.sheetViewAttachmentIds)
+      promises.push(this.handleSheetViewAttachmentIds(response, options.sheetViewAttachmentIds, options.sheetViewViewStateLoadProps));
+    if (options.notLoadedCategoryIds)
+      promises.push(this.handleCategoryIds(response, options.notLoadedCategoryIds));
+    if (options.spatialViewId)
+      promises.push(this.handleSpatialViewId(response, options.spatialViewId, options.spatialViewViewStateLoadProps));
+    if (options.notLoadedModelSelectorStateModels)
+      promises.push(this.handleModelSelectorStateModels(response, options.notLoadedModelSelectorStateModels));
+    if (options.baseModelId)
+      promises.push(this.handleBaseModelId(response, options.baseModelId));
+    await Promise.all(promises);
+    return response;
+  }
+
+  private async handleBaseModelId(response: HydrateViewStateResponseProps, baseModelId: Id64String) {
+    let modelProps;
+    try {
+      modelProps = this._imodel.models.getModelJson({ id: baseModelId });
+    } catch (err) {
+
+    }
+    response.baseModelProps = modelProps;
+  }
+
+  private async handleModelSelectorStateModels(response: HydrateViewStateResponseProps, models: CompressedId64Set) {
+    const decompressedModelIds = CompressedId64Set.decompressSet(models);
+
+    const modelJsonArray: ModelProps[] = [];
+    for (const id of decompressedModelIds) {
+      try {
+        const modelProps = this._imodel.models.getModelJson({ id });
+        modelJsonArray.push(modelProps);
+      } catch (error) {
+        if (decompressedModelIds.size === 1)
+          throw error; // if they're asking for more than one model, don't throw on error.
```

Comment:
Don't throw on backend - just omit invalid model from response.

Reply:
Changed it